web/src/components/organisms/Project/Schema/hooks.ts

```diff
@@ -0,0 +1,172 @@
+import { useCallback, useMemo, useState } from "react";
+
+import { Model } from "@reearth-cms/components/molecules/Schema/types";
+import {
+  useGetModelsQuery,
+  useCreateModelMutation,
+  useCreateFieldMutation,
+  SchemaFiledType,
+  SchemaFieldTypePropertyInput,
+} from "@reearth-cms/gql/graphql-client-api";
+
+type Params = {
+  projectId?: string;
+  modelId?: string;
+};
+
+export default ({ projectId, modelId }: Params) => {
+  const [modelModalShown, setModelModalShown] = useState(false);
+  const [fieldModalShown, setFieldModalShown] = useState(false);
+
+  const { data, refetch } = useGetModelsQuery({
+    variables: { projectId: projectId ?? "", first: 100 },
+    skip: !projectId,
+  });
+
+  const models = useMemo(() => {
+    return (data?.models.nodes ?? [])
+      .map<Model | undefined>(model =>
+        model
+          ? {
+              id: model.id,
+              description: model.description,
+              name: model.name,
+              key: model.key,
+              schema: {
+                id: model.schema.id,
+                fields: model.schema.fields.map(field => ({
+                  id: field.id,
+                  description: field.description,
+                  title: field.title,
+                  type: field.type,
+                  unique: field.unique,
+                  required: field.required,
+                })),
+              },
+            }
+          : undefined,
+      )
+      .filter((model): model is Model => !!model);
+  }, [data?.models.nodes]);
+
+  const rawModel = useMemo(
+    () => data?.models.nodes.find((p: any) => p?.id === modelId),
```

Comment:
Can't p be typed better than any?

Reply:
@KaWaite , it is related to generated files, could not find suitable typing.

Reply:
But you seem to know what p is (here you check if its id is same as modelId and then you use the rawModel to create the modal below) so  can't you create a model type  that meets your expectations and the data you are receiving?

Reply:
Also, `p` doesn't seem like it fits. If it was a plugin, that makes sense. If they are models `m` or `model` is probably clearer